src/main/java/org/jitsi/jibri/JibriManager.kt

```diff
@@ -0,0 +1,203 @@
+
+    /**
+     * Starts a [FileRecordingJibriService] to record the call described
+     * in the params to a file.  Returns a [StartServiceResult] to denote
+     * whether the service was started successfully or not.
+     */
+    @Synchronized
+    fun startFileRecording(
+            serviceParams: ServiceParams,
+            fileRecordingParams: FileRecordingParams,
+            serviceStatusHandler: JibriServiceStatusHandler? = null): StartServiceResult {
+        logger.info("Starting a file recording with params: $fileRecordingParams")
+        if (busy()) {
+            logger.info("Jibri is busy, can't start service")
+            return StartServiceResult.BUSY
+        }
+        val service = FileRecordingJibriService(
+                RecordingOptions(
+                    config.recordingDirectory,
+                    fileRecordingParams.callParams,
+                    config.finalizeRecordingScriptPath
+                )
+            )
+        return startService(service, serviceParams, serviceStatusHandler)
+    }
+
+    /**
+     * Starts a [StreamingJibriService] to capture the call according
+     * to [streamingParams].  Returns a [StartServiceResult] to
+     * denote whether the service was started successfully or not.
+     */
+    @Synchronized
+    fun startStreaming(
+            serviceParams: ServiceParams,
+            streamingParams: StreamingParams,
+            serviceStatusHandler: JibriServiceStatusHandler? = null): StartServiceResult {
+        logger.info("Starting a stream with params: $streamingParams")
+        if (busy()) {
+            logger.info("Jibri is busy, can't start service")
+            return StartServiceResult.BUSY
+        }
+        val service = StreamingJibriService(StreamingOptions(
+                streamingParams.youTubeStreamKey,
+                streamingParams.callParams
+        ))
+        return startService(service, serviceParams, serviceStatusHandler)
+    }
+
+    /**
+     * Helper method to handle the boilerplate of starting a [JibriService].
+     * Returns a [StartServiceResult] to denote whether the service was
+     * started successfully or not.
+     */
+    private fun startService(
+            jibriService: JibriService,
+            serviceParams: ServiceParams,
+            serviceStatusHandler: JibriServiceStatusHandler? = null): StartServiceResult {
+        publishStatus(JibriStatusPacketExt.Status.BUSY)
+        if (serviceStatusHandler != null) {
+            jibriService.addStatusHandler(serviceStatusHandler)
+        }
+        // The manager adds its own status handler so that it can stop
+        // the error'd service and update presence appropriately
+        jibriService.addStatusHandler {
+            when (it) {
+                JibriServiceStatus.ERROR, JibriServiceStatus.FINISHED -> stopService()
+            }
+        }
+
+        if (!jibriService.start()) {
+            return StartServiceResult.ERROR
```

Comment:
Does something at this point need to publish that the status is no longer BUSY?

Reply:
the "busy" state is based on the value of `currentActiveService` which won't get assigned until after this line, so if we return here we never made it to busy.